node/sc/multi_bridge_test.go

```diff
@@ -0,0 +1,354 @@
+
+	acc = info.accounts[1]
+	tx = SendHandleKLAYTransfer(info.b, acc, to, transferAmount, sentNonce, sentBlockNumber, t)
+	info.sim.Commit()
+	assert.NoError(t, bind.CheckWaitMined(info.sim, tx))
+
+	for {
+		select {
+		case ev := <-info.handleCh:
+			assert.Equal(t, nonceOffset, ev.HandleNonce)
+			return
+		case err := <-info.handleSub.Err():
+			t.Fatal("Contract Event Loop Running Stop by sub.Err()", "err", err)
+		case <-time.After(timeOut):
+			t.Fatal("Contract Event Loop Running Stop by timeout")
+		}
+	}
+}
+
+// TestMultiBridgeKLAYTransfer2 checks the following:
+// - failed value transfer without proper transaction counts.
+// - timeout is expected since operator threshold will not be satisfied.
+func TestMultiBridgeKLAYTransfer2(t *testing.T) {
+	info := prepareMultiBridgeEventTest(t)
+	acc := info.accounts[0]
+	to := common.Address{100}
+
+	opts := &bind.TransactOpts{From: acc.From, Signer: acc.Signer, GasLimit: gasLimit}
+	tx, err := info.b.SetOperatorThreshold(opts, voteTypeValueTransfer, 2)
+	assert.NoError(t, err)
+	info.sim.Commit()
+	assert.Nil(t, bind.CheckWaitMined(info.sim, tx))
+
+	nonceOffset := uint64(17)
+	sentNonce := nonceOffset
+	transferAmount := uint64(100)
+	sentBlockNumber := uint64(100000)
+
+	tx = SendHandleKLAYTransfer(info.b, acc, to, transferAmount, sentNonce, sentBlockNumber, t)
+	info.sim.Commit()
+	assert.NoError(t, bind.CheckWaitMined(info.sim, tx))
+
+	for {
+		select {
+		case _ = <-info.handleCh:
+			t.Fatal("unexpected handling of value transfer")
+		case err := <-info.handleSub.Err():
+			t.Fatal("Contract Event Loop Running Stop by sub.Err()", "err", err)
+		case <-time.After(timeOut):
+			// expected timeout
+			return
+		}
+	}
+}
+
+// TestMultiBridgeKLAYTransfer3 checks the following:
+// - tx is actually handled when operator threshold is satisfied.
+// - no double spending is made due to additional tx.
+func TestMultiBridgeKLAYTransfer3(t *testing.T) {
+	info := prepareMultiBridgeEventTest(t)
+	acc := info.accounts[0]
+	to := common.Address{100}
+
+	opts := &bind.TransactOpts{From: acc.From, Signer: acc.Signer, GasLimit: gasLimit}
+	tx, err := info.b.SetOperatorThreshold(opts, voteTypeValueTransfer, 2)
+	assert.NoError(t, err)
+	info.sim.Commit()
+	assert.Nil(t, bind.CheckWaitMined(info.sim, tx))
+
+	nonceOffset := uint64(17)
+	sentNonce := nonceOffset
+	transferAmount := uint64(100)
+	sentBlockNumber := uint64(100000)
+
+	tx = SendHandleKLAYTransfer(info.b, acc, to, transferAmount, sentNonce, sentBlockNumber, t)
+	info.sim.Commit()
+	assert.NoError(t, bind.CheckWaitMined(info.sim, tx))
+
+	acc = info.accounts[1]
+	tx = SendHandleKLAYTransfer(info.b, acc, to, transferAmount, sentNonce, sentBlockNumber, t)
+	info.sim.Commit()
+	assert.NoError(t, bind.CheckWaitMined(info.sim, tx))
+
+	for {
+		select {
+		case _ = <-info.handleCh:
+			acc = info.accounts[2]
+			tx = SendHandleKLAYTransfer(info.b, acc, to, transferAmount, sentNonce, sentBlockNumber, t)
```

Comment:
How can we sure this tx cannot trigger any bridgeHandle event?

Reply:
We check assert.Error()

Reply:
Thank you!